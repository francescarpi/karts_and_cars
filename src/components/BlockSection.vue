<script setup lang="ts">
defineProps({
  title: {
    type: String,
    required: true,
  },
});
</script>

<template>
  <section class="p-4">
    <h3 class="font-default text-xl font-bold w-full border-b-2 pb-2">
      {{ title }}
    </h3>
    <div class="mt-2"><slot /></div>
  </section>
</template>
