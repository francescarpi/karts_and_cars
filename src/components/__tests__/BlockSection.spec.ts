import BlockSection from "../BlockSection.vue";
import { mount } from "@vue/test-utils";
import { describe, it, expect } from "vitest";

describe("BlockSection", () => {
  it("renders properly", () => {
    const wrapper = mount(BlockSection, {
      props: {
        title: "Foo",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
