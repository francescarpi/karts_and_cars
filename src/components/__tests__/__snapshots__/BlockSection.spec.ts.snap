// Vitest Snapshot v1

exports[`BlockSection > renders properly 1`] = `
"<section class=\\"p-4\\">
  <h3 class=\\"font-default text-xl font-bold w-full border-b-2 pb-2 block\\">Foo</h3>
  <div class=\\"mt-4\\"></div>
</section>"
`;
